src/KeyUnit.cpp

```diff
@@ -329,12 +329,11 @@ QString KeyUnit::getKeyName(const Qt::Key keyCode, const Qt::KeyboardModifiers m
         return name % mKeys.value(keyCode);
     }
 
-    return tr("%1undefined key (code: 0x%2)",
+    return QStringLiteral("%1%2").arg(name, tr("undefined key (code: 0x%1)",
```

Comment:
```suggestion
    return qsl("%1%2").arg(name, tr("undefined key (code: 0x%1)",
```

Reply:
Actually we could probably eliminate the `QStringLiteral` that just concatenates two `QString`s entirely now with a:
```diff
-     return qsl("%1%2").arg(name, tr("undefined key (code: 0x%1)",
+     return name % tr("undefined key (code: 0x%1)",
```
which uses the internal `QStringBuilder` class that overloads `operator%()`!